dashboard/test/controllers/omniauth_callbacks_controller_test.rb

```diff
@@ -414,6 +414,85 @@ class OmniauthCallbacksControllerTest < ActionController::TestCase
     assert_equal migrated_student.id, signed_in_user_id
   end
 
+  test 'login: google_oauth2 silently takes over unmigrated student with matching email' do
```

Comment:
I'd probably like to add some more tests here, but I wanted to get this into review sooner rather than later.

Reply:
maybe make a note of the test cases you'd like to cover so we can keep track of them? although that might take as long as just writing the tests 😂

Reply:
Update: The things I really want to cover need to be in integration tests (multiple calls to endpoints), and I'm going to do that in a separate PR, as well as start migrating some of our existing unit tests over.